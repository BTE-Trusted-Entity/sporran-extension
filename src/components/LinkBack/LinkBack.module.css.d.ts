export const __esModule: true;
export const linkBack: string;
