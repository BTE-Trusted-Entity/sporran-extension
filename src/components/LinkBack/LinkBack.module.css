.linkBack {
  composes: buttonIcon from '../../components/Button/Button.module.css';
  position: absolute;
  left: 0;
  top: 0;
  height: 5rem;
  width: 5rem;
  background: url('../images/arrow-left.svg') no-repeat center/auto;
}
