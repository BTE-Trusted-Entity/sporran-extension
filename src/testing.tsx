export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
