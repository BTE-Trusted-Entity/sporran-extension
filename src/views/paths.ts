export { generatePath } from 'react-router-dom';

export const paths = {
  home: '/',
  settings: '/settings',
  access: '/access',
  identity: {
    base: '/identity/',
    add: '/identity/add',
    create: {
      start: '/identity/create',
      backup: '/identity/create/backup',
      verify: '/identity/create/verify',
      password: '/identity/create/password',
    },
    import: {
      start: '/identity/import',
      password: '/identity/import/password',
    },

    overview: '/identity/:address/:type(created|imported|pwreset)?',
    reset: {
      start: '/identity/:address/reset',
      password: '/identity/:address/reset/password',
    },
    send: {
      start: '/identity/:address/send',
      review: '/identity/:address/send/review',
      warning: '/identity/:address/send/warning',
    },
    receive: '/identity/:address/receive',
    credentials: {
      base: '/identity/:address/credentials',
      presentation: '/identity/:address/credentials/:hash/presentation',
    },
    remove: '/identity/:address/remove',
    did: {
      upgrade: {
        start: '/identity/:address/did/upgrade',
        kilt: '/identity/:address/did/upgrade/kilt',
        euro: '/identity/:address/did/upgrade/euro',
      },
      manage: {
        start: '/identity/:address/did',
        downgrade: {
          base: '/identity/:address/did/downgrade',
          warning: {
            web3name: '/identity/:address/did/downgrade/warning/web3name',
            credentials: '/identity/:address/did/downgrade/warning/credentials',
          },
          sign: '/identity/:address/did/downgrade/sign',
        },
        endpoints: {
          start: '/identity/:address/did/endpoints',
          edit: '/identity/:address/did/endpoints/:id?',
          sign: '/identity/:address/did/endpoints/sign',
        },
      },
      repair: '/identity/:address/did/repair',
    },
    web3name: {
      create: {
        base: '/identity/:address/web3name/create',
        info: '/identity/:address/web3name/create/info',
        form: '/identity/:address/web3name/create/form',
        choose: '/identity/:address/web3name/create/choose/:web3name',
        kilt: '/identity/:address/web3name/create/kilt/:web3name',
        euro: '/identity/:address/web3name/create/euro/:web3name',
      },
      manage: {
        start: '/identity/:address/web3name',
        remove: '/identity/:address/web3name/remove',
      },
    },
  },
  popup: {
    base: '/popup',
    import: '/popup/import',
    access: '/popup/access',
    claim: '/identity/:address/claim',
    save: '/popup/save',
    share: {
      start: '/popup/share',
      sign: '/popup/share/sign',
    },
    sign: '/popup/sign',
    signRaw: '/popup/signRaw',
    signDid: {
      start: '/identity/:address/did/sign',
      sign: '/identity/:address/did/sign/password',
      credentials: '/identity/:address/did/sign/credentials',
    },
    signDidExtrinsic: '/identity/:address/did/signExtrinsic',
    createDid: '/identity/:address/did/create',
  },
};
