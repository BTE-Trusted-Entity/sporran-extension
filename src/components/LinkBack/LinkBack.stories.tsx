import { Meta } from '@storybook/react';
import { Link, MemoryRouter, Route } from 'react-router-dom';

import * as styles from './LinkBack.module.css';

import { generatePath, paths } from '../../views/paths';

import { LinkBack } from './LinkBack';

export default {
  title: 'Components/LinkBack',
} as Meta;

export function CSS(): JSX.Element {
  return <Link to="" title="Back" className={styles.linkBack} />;
}

export function AsButton(): JSX.Element {
  return <LinkBack />;
}

export function AsLink(): JSX.Element {
  return (
    <MemoryRouter
      initialEntries={[
        generatePath(paths.identity.send.start, { address: 'FOO' }),
      ]}
    >
      <Route path={paths.identity.send.start}>
        <LinkBack to={paths.identity.overview} />
      </Route>
    </MemoryRouter>
  );
}
