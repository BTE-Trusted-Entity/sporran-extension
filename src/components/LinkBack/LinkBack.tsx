import { Link, useHistory, useParams } from 'react-router-dom';
import { browser } from 'webextension-polyfill-ts';

import * as styles from './LinkBack.module.css';

import { generatePath } from '../../views/paths';

interface Props {
  to?: string;
}

export function LinkBack({ to }: Props): JSX.Element {
  const t = browser.i18n.getMessage;

  const { goBack } = useHistory();
  const params = useParams();

  if (to) {
    return (
      <Link
        title={t('common_action_back')}
        aria-label={t('common_action_back')}
        to={generatePath(to, params)}
        className={styles.linkBack}
      />
    );
  }

  return (
    <button
      type="button"
      title={t('common_action_back')}
      aria-label={t('common_action_back')}
      onClick={goBack}
      className={styles.linkBack}
    />
  );
}
