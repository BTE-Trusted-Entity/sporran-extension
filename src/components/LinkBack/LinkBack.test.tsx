import { MemoryRouter, Route } from 'react-router-dom';

import { render } from '../../testing/testing';
import { generatePath, paths } from '../../views/paths';

import { LinkBack } from './LinkBack';

describe('LinkBack', () => {
  it('should render as button', async () => {
    const { container } = render(<LinkBack />);
    expect(container).toMatchSnapshot();
  });

  it('should render as link', async () => {
    const { container } = render(
      <MemoryRouter
        initialEntries={[
          generatePath(paths.identity.send.start, { address: 'FOO' }),
        ]}
      >
        <Route path={paths.identity.send.start}>
          <LinkBack to={paths.identity.overview} />
        </Route>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
