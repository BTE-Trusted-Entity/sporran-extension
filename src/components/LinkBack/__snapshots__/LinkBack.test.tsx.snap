// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkBack should render as button 1`] = `
<div>
  <button
    aria-label="Back"
    class="linkBack"
    title="Back"
    type="button"
  />
</div>
`;

exports[`LinkBack should render as link 1`] = `
<div>
  <a
    aria-label="Back"
    class="linkBack"
    href="/identity/FOO"
    title="Back"
  />
</div>
`;
